x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx

```diff
@@ -366,32 +377,58 @@ export const AgentUpgradeAgentModal: React.FunctionComponent<AgentUpgradeAgentMo
           />
         ) : isSingleAgent ? (
-          <>
-            <p>
+          !isAgentUpgradeable(agents[0], latestAgentVersion || '', selectedVersion[0].value) ? (
+            <EuiCallOut
+              data-test-subj="agentUpgradeModal.notUpgradeableCallout"
+              color="warning"
+              iconType="warning"
+              title={
+                <FormattedMessage
+                  id="xpack.fleet.upgradeAgents.notUpgradeable"
+                  defaultMessage="The selected agent is not upgradeable."
+                />
+              }
+            >
               <FormattedMessage
-                id="xpack.fleet.upgradeAgents.upgradeSingleDescription"
-                defaultMessage="This action will upgrade the agent running on '{hostName}' to version {version}. This action can not be undone. Are you sure you wish to continue?"
+                id="xpack.fleet.upgradeAgents.notUpgradeableMsg"
+                defaultMessage="Reason: {reason}"
                 values={{
-                  hostName: ((agents[0] as Agent).local_metadata.host as any).hostname,
-                  version: getVersion(selectedVersion),
+                  reason: getNotUpgradeableMessage(
+                    agents[0],
+                    latestAgentVersion,
+                    selectedVersion[0].value
+                  ),
                 }}
               />
-            </p>
-            {isUpdating && (
+            </EuiCallOut>
+          ) : (
+            <>
               <p>
-                <em>
-                  <FormattedMessage
-                    id="xpack.fleet.upgradeAgents.upgradeSingleTimeout"
-                    // TODO: Add link to docs regarding agent upgrade cooldowns
-                    defaultMessage="Note that you may only restart an upgrade every {minutes} minutes to ensure that the upgrade will not be rolled back."
-                    values={{
-                      minutes: AGENT_UPGRADE_COOLDOWN_IN_MIN,
-                    }}
-                  />
-                </em>
+                <FormattedMessage
+                  id="xpack.fleet.upgradeAgents.upgradeSingleDescription"
+                  defaultMessage="This action will upgrade the agent running on '{hostName}' to version {version}. This action can not be undone. Are you sure you wish to continue?"
+                  values={{
+                    hostName: ((agents[0] as Agent).local_metadata.host as any).hostname,
+                    version: getVersion(selectedVersion),
+                  }}
+                />
               </p>
-            )}
-          </>
+              {isUpdating && (
+                <p>
+                  <em>
+                    <FormattedMessage
+                      id="xpack.fleet.upgradeAgents.upgradeSingleTimeout"
+                      // TODO: Add link to docs regarding agent upgrade cooldowns
```

Comment:
do we have an issue for this in ingest-docs?

Reply:
Not sure, the comment was already there, I tried to find docs for it but couldn't. I can open an issue for it